[REF] product: Set max iterations to look for product combinations

moylop260 opened the pull request:
Sometimes the products are stuck for a long time looking for combinations

This patch set max iterations for this infinite loop

It is reproduced with the following script

```python
self.env.cr.execute("""
    SELECT ptal.product_tmpl_id,
        ARRAY_AGG(ptav.id) AS ptav2del
    FROM product_template_attribute_line AS ptal
    JOIN product_attribute AS pa
      ON ptal.attribute_id = pa.id
    JOIN product_template_attribute_value AS ptav
      ON ptal.id = ptav.attribute_line_id
    WHERE pa.create_variant = 'always'
      AND ptav.id IS NOT NULL
      AND ptal.active IS TRUE
    GROUP BY ptal.product_tmpl_id
""")
res = self.env.cr.fetchall()

for r in res:
    product_tmpl = self.env["product.template"].browse(r[0])
    product_templ_attr_vals = self.env["product.template.attribute.value"].browse(r[1])
    product_templ_attr_vals.unlink()
    print(product_tmpl)
    combination = product_tmpl._get_first_possible_combination()
```

luisg123v commented:
Maybe this could be a fix to Odoo.

moylop260 commented:
> Maybe this could be a fix to Odoo.

I thought that, too

However, we need to justify it very well, and the current way to reproduce it is by deleting the `product_template_attribute_value` records and it is not reproduced functionally  

The first impression looks like a migration issue for this database not reproducible from Odoo itself

But we keep checking...

BTW could you check the migration script used for this case, please?

hugho-ad commented:
What about ese similar technique than the dynamic_variant_limit?
https://github.com/Vauxoo/odoo/blob/8a64fddb7628053f74b4a2a9c15dacd8b7f4f9f7/addons/product/models/product_template.py#L783

moylop260 committed:
[REF] product: Set max iterations to look for product combinations

Sometimes the products are stuck for a long time looking for combinations

This patch set max iterations for this infinite loop
